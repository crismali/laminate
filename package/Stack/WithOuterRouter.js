import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { WrapRouter } from './WrapRouter'

export class OuterHistory extends Component {
  static propTypes = {
    children: PropTypes.any.isRequired
  }

  static contextTypes = {
    outerRouter: PropTypes.object.isRequired
  }

  render () {
    return <WrapRouter router={this.context.outerRouter}>
      {this.props.children}
    </WrapRouter>
  }
}
