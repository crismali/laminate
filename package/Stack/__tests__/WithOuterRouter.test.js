/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { WithOuterRouter } from '../WithOuterRouter'

const Child = () => 'a child'

it('renders its children with a wrapped router that is the outerRouter from context', () => {
  const subject = shallow(<WithOuterRouter>
    <Child />
  </WithOuterRouter>, { context: { outerRouter: { the: 'outer router' } } })
  expect(subject).toMatchSnapshot()
})
