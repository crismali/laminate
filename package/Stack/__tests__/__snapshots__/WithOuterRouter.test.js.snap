// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders its children with a wrapped router that is the outerRouter from context 1`] = `
<WrapRouter
  router={
    Object {
      "the": "outer router",
    }
  }
>
  <Child />
</WrapRouter>
`;
